ddl/partition.go

```diff
@@ -1212,3 +1214,56 @@ func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo)
 	tblInfo.Partition.Definitions = newDefs
 	return nil
 }
+
+func checkPlacementRules(t *meta.Meta, schemaID, tblID, partitionID int64, rules []*placement.Rule) ([]*placement.Rule, error) {
+	for _, rule := range rules {
+		randID, err := t.GenAutoRandomID(schemaID, tblID, 1)
```

Comment:
It is changed to jobID + index in the next commit. I want to keep the ID same.

 In case that TiDB sent the request to PD, but TiDB is down suddenly. When retrying the job, it may assign a new ID for each rule, such that it duplicates rules.

Reply:
Rule ID can be allocated when creating the job. Then the rule ID will be stored in the job. When retrying the job, the rule ID will be decoded.
It's also fine to encode it with job ID. But I think the schema ID is unnecessary.

Reply:
I want to delete all rules related to the database, simply by this schema ID prefix.

Reply:
When will we delete all rules related to a database?
You can't infer the source of the rule from `id`. That is, you don't know whether this rule was defined on the table itself, or was inherited from its database.
So even when you're  changing the rule of a database, the database ids are unnecessary.
The only case I can think of is when you're dropping a database. But the table ids can still be fetched from metadata.

Reply:
> when will we..

Alter/Drop may need to delete old rules.

> you don't know whether this rule was defined on the table itself, or was inherited from its database.

That is simple. Just make the id format different. You can mark anything you want in this string.

 > But the table ids can still be fetched from metadata.

Yes, I can perform that without database id. But the thing is that prefix matching needs to go through all rules(on PD side). So I would like to have as fewer as possible deletions.

Anyway, it is not so clear that what ID format should I choose.

Reply:
>Alter/Drop may need to delete old rules.
Alter/Drop what? a database? Any examples?

Reply:
Alter is an overriding operation. But ID is different, so it is actually not overriding. One way is to delete old rules with specific prefix first, then define new rules.

Reply:
Alter what? database or table?

Reply:
Both. For a table, you need to override old rules. For a database, you may need to override inherited rules.

> Because all the placement rules on t1 are inherited from db, they will keep the same with db all the time. 

This statement introduced a problem on synchronization.